dsgrid/registry/registry_database.py

```diff
@@ -1,415 +1,643 @@
+        new_model = self._add_database_id(conn, table, db_id)
+        self._set_current_version(conn, model_type, model_id, db_id)
+        logger.debug("Inserted model_type=%s model_id=%s id=%s", model_type, model_id, db_id)
+        return new_model
+
+    def update_model(
+        self,
+        conn: Connection,
+        model_type: RegistryType,
+        model: dict[str, Any],
+        registration: RegistrationModel,
+    ) -> dict[str, Any]:
+        """Add a model to the database. Sets the id field of the model with the database value."""
+        table = self.get_table(RegistryTables.MODELS)
+        model_id = self._get_model_id(model_type, model)
+        res = conn.execute(
+            insert(table).values(
+                registration_id=registration.id,
+                model_type=model_type.value,
+                model_id=model_id,
+                version=model["version"],
+                model=model,
+            )
         )
-        count = cursor.count()
-        assert count <= 1, f"{root_db_id=} {count=}"
-        if count == 0:
-            raise DSGValueNotRegistered(f"{root_db_id=} is not registered")
-        model_db_id = cursor.next()
-        if model_db_id is None:
-            raise DSGValueNotRegistered(f"{root_db_id=} is not registered")
-        return self.get_version(model_db_id)
-
-    def _get_by_version(self, db_id, version):
-        cursor = self._db.aql.execute(
-            f"""
-            FOR v, e, p in 0..{MAX_CONFIG_VERSIONS}
-                OUTBOUND "{db_id}"
-                GRAPH "{GRAPH}"
-                OPTIONS {{edgeCollections: ["{Edge.UPDATED_TO.value}"]}}
-                FILTER e.version == "{version}"
-                RETURN p.vertices[-1]
-        """,
-            count=True,
+        db_id = res.lastrowid
+        new_model = self._add_database_id(conn, table, db_id)
+        self._update_current_version(conn, model_type, model_id, db_id)
+        logger.error("Updated model_type=%s model_id=%s id=%s", model_type, model_id, db_id)
+        return new_model
+
+    def _add_database_id(self, conn: Connection, table: Table, db_id: int) -> dict[str, Any]:
+        stmt = select(table.c.model).where(table.c.id == db_id)
+        res = conn.execute(stmt).fetchone()
+        assert res
+        data = res[0]
+        # Ensure that all future reads will have the database ID set in the model.
+        data["id"] = db_id
+        conn.execute(update(table).where(table.c.id == db_id).values(model=data))
+        return data
+
+    def _set_current_version(
+        self, conn: Connection, model_type: RegistryType, model_id: str, db_id: int
+    ) -> None:
+        table = self.get_table(RegistryTables.CURRENT_VERSIONS)
+        conn.execute(
+            insert(table).values(
+                model_type=model_type.value,
+                model_id=model_id,
+                current_id=db_id,
+                update_timestamp=str(datetime.now()),
+            )
```

Comment:
This is for new `model_id`s? (It's not deleting previous entries for `model_id`.)